src/libwalletqt/Wallet.cpp

```diff
@@ -662,6 +662,26 @@ QString Wallet::getTxKey(const QString &txid) const
   return QString::fromStdString(m_walletImpl->getTxKey(txid.toStdString()));
 }
 
+void Wallet::getTxKeyAsync(const QString &txid, const QJSValue &ref) const
+{
+    QFuture<QString> future = QtConcurrent::run(this, &Wallet::getTxKey, txid);
+    auto watcher = new QFutureWatcher<QString>();
+
+    connect(watcher, &QFutureWatcher<QString>::finished,
+            this, [this, watcher, txid, ref]() {
+          QFuture<QString> future = watcher->future();
+          watcher->deleteLater();
+
+          auto txKey = future.result();
+          if (ref.isCallable()){
+              QJSValue cb(ref);
+              cb.call(QJSValueList {txid, txKey});
+          }
+          emit txKeyReceived(txid, txKey, ref);
```

Comment:
For greater flexilibity I've added standard signal which is triggered anyway. The last parameter can be aux value (e.g., UI element, related information) if it is not used as a callback.

Reply:
Think we won't ever need this. Callbacks are flexible enough